File: sorts/bubble_sort/bubble_sort.py
def bubble_sort(li):
  for i in range(len(li)):
    for j in range(i+1, len(li)):
      if li[j] < li[i]:
        tmp = li[i]
        li[i] = li[j]
        li[j] = tmp

li = [8, 5, 2, 3, 1, 9, 4, 6, 7, 0]
print(li)
bubble_sort(li)
print(li)

File: sorts/bubble_sort/bubble_sort.c
#include <stdio.h>

void bubbleSort(int*, int);
void printNumbers(int*, int);

void bubbleSort(int* arr, int n){
  for (int i=0; i < n; i++) {
    for (int j=i+1; j < n; j++) {
      if (arr[j] < arr[i]) {
        int tmp = arr[i];
        arr[i] = arr[j];
        arr[j] = tmp;
      }
    }
  }
}

void printNumbers(int* arr, int n) {
  for (int i=0; i < n; i++) {
    printf("%d ", arr[i]);
  }
  printf("\n");
}

int main() {
  int arr[10] = {8, 5, 2, 3, 1, 9, 4, 6, 7, 0};
  int n = sizeof(arr) / sizeof(arr[0]);
  printNumbers(arr, n);
  bubbleSort(arr, n);
  printNumbers(arr, n);
}

